Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift

```diff
@@ -206,11 +206,17 @@ struct RenderContentCompiler: MarkupVisitor {
     }
     
     mutating func resolveTopicReference(_ destination: String) -> ResolvedTopicReference? {
-        if let cached = context.referenceIndex[destination] {
+         if let cached = context.referenceIndex[destination] {
+            if let node = context.topicGraph.nodeWithReference(cached), !context.topicGraph.isLinkable(node.reference) {
```

Comment:
Why is the change here needed? Was it just missing before?

Reply:
I _thought_ that it was needed because the code below does it but now that I think about it I also think that the link resolution would have already done this

Reply:
Actually, only `MarkupReferenceResolver` filters references that are linkable so I guess it's possible for the context to have resolved and saved a reference that isn't linkable.

So in order for the cached and non-cached code paths to behave the same, the cached code path should also check that the reference is linkable.